hpx/runtime/threads/thread_helpers.hpp

```diff
@@ -104,11 +105,24 @@ namespace hpx { namespace threads
     ///                   of hpx#exception.
     HPX_API_EXPORT thread_id_type set_thread_state(thread_id_type const& id,
         util::steady_time_point const& abs_time,
+        std::atomic<bool>& started,
         thread_state_enum state = pending,
         thread_state_ex_enum stateex = wait_timeout,
         thread_priority priority = thread_priority_normal,
         error_code& ec = throws);
 
+    inline thread_id_type set_thread_state(thread_id_type const& id,
+        util::steady_time_point const& abs_time,
+        thread_state_enum state = pending,
+        thread_state_ex_enum stateex = wait_timeout,
+        thread_priority priority = thread_priority_normal,
+        error_code& ec = throws)
+    {
+        std::atomic<bool> timer_started(false);
```

Comment:
Won't this atomic go out of scope before the timer thread can access it?

Reply:
Good catch!